### ISSUE TITLE
[BUG] advance daemonset can not undo to previous version

### ISSUE THREAD

**Issue by losha228** [openkruise/kruise#1100]
**What happened**:
I am testing advanced daemonset, I can see rollout history for the ads,  when I rollback to version3, it works well, but when rollback to version2, it said not found for clonesets,  it is obvious it is not a clonesets.
```
$ kubectl kruise rollout history ads/nginx-ds-kruise 
+ kubectl kruise rollout history ads/nginx-ds-kruise
daemonset.apps.kruise.io/nginx-ds-kruise 
REVISION  CHANGE-CAUSE
2         <none>
3         <none>

$ kubectl kruise rollout undo  ads/nginx-ds-kruise --to-revision=3
+ kubectl kruise rollout undo ads/nginx-ds-kruise --to-revision=3
daemonset.apps.kruise.io/nginx-ds-kruise skipped rollback (current template already matches revision 3)

$ kubectl kruise rollout undo  ads/nginx-ds-kruise --to-revision=2
+ kubectl kruise rollout undo ads/nginx-ds-kruise --to-revision=2
error: failed restoring revision 2: clonesets.apps.kruise.io "nginx-ds-kruise" not found
```

**What you expected to happen**:
daemonset can support undo

**How to reproduce it (as minimally and precisely as possible)**:
1.  rollout an ads for several versions
2.  rollback to previous version

**Anything else we need to know?**:

**Environment**:
- Kruise version:
```azureuser@master:~/test/scheduler-losha$ kubectl kruise version                                          
+ kubectl kruise version
Client Version: version.Info{Major:"1", Minor:"0", GitVersion:"v1.0.4", GitCommit:"e1b55f7d11d906edf31a80ba695bc8857b881b75", GitTreeState:"clean", BuildDate:"2022-07-22T11:51:37Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.6", GitCommit:"b39bf148cd654599a52e867485c02c4f9d28b312", GitTreeState:"clean", BuildDate:"2022-09-21T13:12:04Z", GoVersion:"go1.18.6", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.0) and server (1.24) exceeds the supported minor version skew of +/-1
 ```
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"0", GitVersion:"v1.0.4", GitCommit:"e1b55f7d11d906edf31a80ba695bc8857b881b75", GitTreeState:"clean", BuildDate:"2022-07-22T11:51:37Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.6", GitCommit:"b39bf148cd654599a52e867485c02c4f9d28b312", GitTreeState:"clean", BuildDate:"2022-09-21T13:12:04Z", GoVersion:"go1.18.6", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.0) and server (1.24) exceeds the supported minor version skew of +/-1
$ kubectl version
+ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.6", GitCommit:"b39bf148cd654599a52e867485c02c4f9d28b312", GitTreeState:"clean", BuildDate:"2022-09-21T13:19:24Z", GoVersion:"go1.18.6", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.6", GitCommit:"b39bf148cd654599a52e867485c02c4f9d28b312", GitTreeState:"clean", BuildDate:"2022-09-21T13:12:04Z", GoVersion:"go1.18.6", Compiler:"gc", Platform:"linux/amd64"}
```

- Install details (e.g. helm install args):
```helm install kruise openkruise/kruise --version 1.3.0```
- Others:

**Comment by FillZpp** [comment]
@losha228 Thanks for reporting. This must be a bug of [openkruise/kruise-tools](https://github.com/openkruise/kruise-tools) and I'm going to fix it.

**Comment by FillZpp** [comment]
@losha228 It has fixed in the latest kubectl-kruise [v1.0.5](https://github.com/openkruise/kruise-tools/releases/tag/v1.0.5). You can upgrade the command-line tool using krew or manually.
